### PR TITLE
Resize to new size. Fixes #788

### LINKED ISSUE #788: sizehint! ineffective for RobinDict
```
julia> function check_alloc(d, n)
           for i in 1:n
               a = @allocated d[i] = i
               println(i, " ", a)
           end
           nothing
       end
check_alloc (generic function with 1 method)

julia> check_alloc(sizehint!(RobinDict{Int, Int}(), 16), 16)
1 0
2 0
3 0
4 0
5 0
6 0
7 0
8 0
9 0
10 0
11 0
12 0
13 1584
14 0
15 0
16 0
```

The root cause is here https://github.com/JuliaCollections/DataStructures.jl/blob/f90dd8ceb1808123136d96599fbf41407ef4fadc/src/robin_dict.jl#L200-L201 which should instead be:
```
        resize!(h.keys, newsz)
        resize!(h.vals, newsz)
```

### PR DESCRIPTION
**Pull request by dantaras**
Let me know if you think a test is required here. I didn't add one as the only side effects are extra allocations in some cases which wouldn't make for a particularly reliable test.

### CODE REVIEW

**Comment by dantaras** [comment]
Any chance we can get this reviewed and merged in soon? It's quite a straightforward fix.

**Review by eulerkochy — APPROVED** [review]

**Comment by oxinabox** [comment]
@eulerkochy you have merge rights and should merge this if you are happy with it.
And backport it, if required.

**Pull request merged**
